code/modules/overmap/sectors.dm

```diff
@@ -208,6 +208,9 @@ var/global/list/known_overmap_sectors
 	closeToolTip(usr) //No reason not to, really
 	..()
 
+/obj/effect/overmap/visitable/proc/get_topmost_level_data()
+	return SSmapping.levels_by_z[map_z[1]]
```

Comment:
You're right about that. Currently the topmost level is usually the highest z in the sector's zlevels list because of how z stacks works currently. 

So, in short, easy way to fix this is to do ``max(map_z)`` instead of ``map_z[1]``. But, if we rework z stacks to have non-sequential up/down level connections eventually, that wouldn't work. 
It would be nice to have a standard way of accessing z stacks we could use instead of taking shortcuts like I did. Currently it's mostly done through ``global.z_levels`` and the ``/obj/abstract/map_data`` object, but those are kind of subject to change with z mimic ports.